op-dispute-mon/mon/withdrawals.go

```diff
@@ -0,0 +1,46 @@
+package mon
+
+import (
+	"github.com/ethereum-optimism/optimism/op-dispute-mon/mon/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"golang.org/x/exp/maps"
+)
+
+type WithdrawalMetrics interface {
+	RecordWithdrawalRequests(delayedWeth common.Address, matches bool, count int)
+}
+
+type WithdrawalMonitor struct {
+	logger  log.Logger
+	metrics WithdrawalMetrics
+}
+
+func NewWithdrawalMonitor(logger log.Logger, metrics WithdrawalMetrics) *WithdrawalMonitor {
+	return &WithdrawalMonitor{
+		logger:  logger,
+		metrics: metrics,
+	}
+}
+
+func (w *WithdrawalMonitor) CheckWithdrawals(games []*types.EnrichedGameData) {
+	for _, game := range games {
+		w.validateGameWithdrawals(game)
+	}
```

Comment:
This doesn't actually work - games may share a `DelayedWETH` contract so we need to ensure we sum up the values for each game that uses the same `DelayedWETH` - currently we only report requests for the last game that uses a WETH contract because each game overwrites the values for the one before it.  Probably needs to work more like how bond collateral checks do where it runs through all the games computing values and then just updates metrics with one call at the end (https://github.com/ethereum-optimism/optimism/blob/ac75a70d8e14847562ac123b6f7862e8c0f1b84b/op-dispute-mon/mon/bonds/monitor.go#L27-L32)